File: bindings/java/src/main/java/systems/zlink/contracts/sockets/SocketEnums/SubmitResult.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.contracts.sockets;


/** The outcome of submitting a send or publish. */
public enum SubmitResult {
    OK(0),
    BACKPRESSURED(1),
    NOT_CONNECTED(2),
    NOT_FOUND(3),
    TERMINATED(4),
    INVALID_HANDLE(5),
    INVALID_ARGUMENT(6),
    NOT_SUPPORTED(7),
    INVALID_STATE(8),
    THREAD_VIOLATION(9),
    OUT_OF_MEMORY(10),
    SEQ_EXHAUSTED(11),
    INTERNAL_ERROR(12),
    NOT_ADMITTED(13);

    private final int value;

    SubmitResult(int value) {
        this.value = value;
    }

    public int value() {
        return value;
    }

    private static final SubmitResult[] VALUES = values();

    public static SubmitResult fromValue(int value) {
        for (SubmitResult result : VALUES) {
            if (result.value == value) {
                return result;
            }
        }
        throw new IllegalArgumentException("invalid SubmitResult value: " + value);
    }
}

File: bindings/dotnet/src/Zlink/Contracts/Errors/SubmitResult.cs
// SPDX-License-Identifier: MPL-2.0

namespace Systems.Zlink;

/// <summary>
///     The outcome reported when a socket operation is submitted.
/// </summary>
public enum SubmitResult
{
    /// <summary>The submit succeeded.</summary>
    Ok = 0,

    /// <summary>The target mailbox is back-pressured.</summary>
    Backpressured = 1,

    /// <summary>The target is not connected.</summary>
    NotConnected = 2,

    /// <summary>The target was not found.</summary>
    NotFound = 3,

    /// <summary>The target peer is not admitted into the mesh.</summary>
    NotAdmitted = 13,

    /// <summary>The context or node was terminated.</summary>
    Terminated = 4,

    /// <summary>The handle was null or invalid.</summary>
    InvalidHandle = 5,

    /// <summary>An argument was invalid.</summary>
    InvalidArgument = 6,

    /// <summary>The operation is not supported.</summary>
    NotSupported = 7,

    /// <summary>The lifecycle state rejected the submit.</summary>
    InvalidState = 8,

    /// <summary>The call violated the owning thread affinity.</summary>
    ThreadViolation = 9,

    /// <summary>Out of memory.</summary>
    OutOfMemory = 10,

    /// <summary>The operation sequence space is exhausted.</summary>
    SeqExhausted = 11,

    /// <summary>An internal error occurred.</summary>
    InternalError = 12
}
